QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,845 @@
+                        break;
+
+                    case OrderType.StopMarket:
+                        var stopMarketOrder = (StopMarketOrder)order;
+                        orderPlacement = Client.PlaceOrder(
+                            _accountId,
+                            SymbolMapper.GetBrokerageSymbol(order.Symbol),
+                            ExanteOrderType.Stop,
+                            orderSide,
+                            quantity,
+                            orderDuration,
+                            stopPrice: stopMarketOrder.StopPrice,
+                            gttExpiration: goodTilDateTimeInForceExpiration
+                        );
+                        break;
+
+                    case OrderType.StopLimit:
+                        var stopLimitOrder = (StopLimitOrder)order;
+                        orderPlacement = Client.PlaceOrder(
+                            _accountId,
+                            SymbolMapper.GetBrokerageSymbol(order.Symbol),
+                            ExanteOrderType.Stop,
+                            orderSide,
+                            quantity,
+                            orderDuration,
+                            limitPrice: stopLimitOrder.LimitPrice,
+                            stopPrice: stopLimitOrder.StopPrice,
+                            gttExpiration: goodTilDateTimeInForceExpiration
+                        );
+                        break;
+
+                    default:
+                        throw new NotSupportedException(
+                            $"ExanteBrokerage.ConvertOrderType: Unsupported order type: {order.Type}");
+                }
+
+                foreach (var o in orderPlacement.Data)
+                {
+                    _orderMap[o.OrderId] = order;
+                    order.BrokerId.Add(o.OrderId.ToString());
+                }
+
+                if (!orderPlacement.Success)
+                {
+                    var errorsJson =
+                        JArray.Parse(orderPlacement.Error?.Message ?? throw new InvalidOperationException());
+                    var errorMsg = string.Join(",", errorsJson.Select(x => x["message"]));
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, -1, errorMsg));
+                }
+
+                orderPlacementSuccess = orderPlacement.Success;
+            });
+
+            return orderPlacementSuccess;
+        }
+
+        /// <summary>
+        /// Updates the order with the same id
+        /// </summary>
+        /// <param name="order">The new order information</param>
+        /// <returns>True if the request was made for the order to be updated, false otherwise</returns>
+        public override bool UpdateOrder(Order order)
+        {
+            if (order == null)
+            {
+                throw new ArgumentNullException(nameof(order));
+            }
+
+            var updateResult = true;
+            foreach (var bi in order.BrokerId.Skip(1))
```

Comment:
The only scenario (confirmed by Exante support) when REST API creates multiple orders in response is when placing a limit order with take-profit and stop-loss (one order for the limit order, one connected stop order if any, and one connected limit order with a price if any). It looks as follows in the terminal:

![image](https://user-images.githubusercontent.com/574267/139499205-044ce89e-36da-4b97-9ac2-902c2b1ca0d6.png)

The API call to reproduce it is as follows:

```c#
exanteClient.PlaceOrderAsync(accountId, "ETH.USD", ExanteOrderType.Limit, ExanteOrderSide.Buy,
            quantity: 1m, ExanteOrderDuration.GoodTillCancel, limitPrice: 4160m, 
            takeProfit: 4300m, stopLoss: 4100).Result;
```

And since LEAN doesn't support such kind of orders (AFAIK), I handle only the scenario when order placement produces only a single order.

Reply:
Added comments at becac1dd0c848779251af45d96ac6e881d542919. Feel free to close if you're convinced.